### PR TITLE
Unit tests for vehicle speed.cpp #3081

### PR DESCRIPTION
**Pull request by alxrMironov**
I has found, what wrapping 
`
	CONFIG(vehicleSpeedSensorInputPin) = GPIOA_0;
	brain_pin_markUsed(GPIOA_0, "VSS" PASS_ENGINE_PARAMETER_SUFFIX);
	engineConfiguration->vehicleSpeedCoef = speedCoef;
`
into function causes big mess and tons of parameters, so i just copy-paste it.

Idk why `brain_pin_markUsed()` dont works, and vehicleSpeed still has `NOT_ASSIGNED`, so i used direct 'CONFIG(vehicleSpeedSensorInputPin) = GPIOA_0;' too.

### CODE REVIEW

File: unit_tests/efifeatures.h [hunk 1]
```diff
@@ -76,3 +76,5 @@
 #define EFI_FUEL_PUMP TRUE
 
 #define EFI_LUA TRUE
+
+#define EFI_VEHICLE_SPEED TRUE
```
Comment (rusefillc) [inline unit_tests/efifeatures.h:80]: we already have it in https://github.com/rusefi/rusefi/blob/master/unit_tests/efifeatures.h#L16

File: unit_tests/tests/test_vehicle_speed.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	constexpr float speedCoef = 0.5f;
+
+	// Init global variables
+	CONFIG(vehicleSpeedSensorInputPin) = anyPin;
+	brain_pin_markUsed(anyPin, vehicleSpeedSensorMessage PASS_ENGINE_PARAMETER_SUFFIX);
+	engineConfiguration->vehicleSpeedCoef = speedCoef;
+
+	simulatePeriodicSignalForCallback(eth, freq, vsCallback PASS_ENGINE_PARAMETER_SUFFIX);
+	float measuredSpeed = getVehicleSpeed(PASS_ENGINE_PARAMETER_SIGNATURE);
+	EXPECT_EQ(returnedOnInvalidFreq, measuredSpeed);
```
Comment (rusefillc) [inline unit_tests/tests/test_vehicle_speed.cpp:72]: we cannot use EQ for "float" type, float type is rarely equal. We use ASSERT_NEAR see other usages of ASSERT_NEAR

that's due to nature of float and double

File: unit_tests/tests/test_vehicle_speed.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+TEST(VehicleSpeedSensor, testValidSpeedDetection) {
+	WITH_ENGINE_TEST_HELPER(TEST_ENGINE);
+
+	constexpr float expectedSpeed = 15.0f;
+	constexpr float speedCoef = 0.5f;
+
+	// Init global variables
+	CONFIG(vehicleSpeedSensorInputPin) = anyPin;
+	brain_pin_markUsed(anyPin, vehicleSpeedSensorMessage PASS_ENGINE_PARAMETER_SUFFIX);
```
Comment (mck1117) [inline unit_tests/tests/test_vehicle_speed.cpp:48]: you probably don't need `brain_pin_markUsed`

File: unit_tests/tests/test_vehicle_speed.cpp [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	EXPECT_NEAR(expectedSpeed, measuredSpeed, 0.01);
+
+}
+
+TEST(VehicleSpeedSensor, testInvalidSpeed) {
+	WITH_ENGINE_TEST_HELPER(TEST_ENGINE);
+
+	constexpr float freq = 0.5f; // Invalid period > 1sec
+	constexpr float returnedOnInvalidFreq = 0.0f;
+	constexpr float speedCoef = 0.5f;
```
Comment (mck1117) [inline unit_tests/tests/test_vehicle_speed.cpp:63]: why so many extra constant variables that are used only once?

File: unit_tests/tests/test_vehicle_speed.cpp (hunk 2 above)
Comment (mck1117) [reply unit_tests/tests/test_vehicle_speed.cpp:72]: you can however use `EXPECT_FLOAT_EQ`: it checks that the floats are equal within +-4 LSB

see https://github.com/google/googletest/blob/master/docs/reference/assertions.md#floating-point-comparison-floating-point

File: unit_tests/tests/test_vehicle_speed.cpp (hunk 4 above)
Comment (mck1117) [reply unit_tests/tests/test_vehicle_speed.cpp:63]: These make is much harder to read.  For example, the last two lines of this test would be much clearer if written as

```
// Invalid (slow) interval, should return 0 speed
EXPECT_EQ(0, getVehicleSpeed(PASS_ENGINE_PARAMETER_SIGNATURE));
```

File: unit_tests/tests/test_vehicle_speed.cpp [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+ * 	to simulate periodic signal on input pin with passed callback.
+ */
+static void simulatePeriodicSignalForCallback(
+	EngineTestHelper& eth,
+	float freqHz,
+	vss_callback_fp cb
+	DECLARE_ENGINE_PARAMETER_SUFFIX)
+{
+	constexpr auto periods = 50;
+	auto period = (1 / freqHz);
```
Comment (mck1117) [inline unit_tests/tests/test_vehicle_speed.cpp:29]: Good use of `auto`!

**Pull request merged**